ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/site/js/formcontainerview.js

```diff
@@ -30,23 +30,32 @@
             let self = this;
             this._model.subscribe((action) => {
                 let state = action.target.getState();
-                let body = action.payload?.body;
-                if (body) {
-                    if (body.redirectUrl) {
-                        window.location.href = encodeURI(body.redirectUrl);
-                    } else if (body.thankYouMessage) {
-                        let formContainerElement = document.querySelector("[data-cmp-path='"+ self._path +"']");
-                        let thankYouMessage = document.createElement("div");
-                        thankYouMessage.setAttribute("class", "tyMessage");
-                        thankYouMessage.innerHTML = body.thankYouMessage;
-                        formContainerElement.replaceWith(thankYouMessage);
-                    }
+                // execute the handler only if there are no rules configured on submitSuccess event.
+                if (!state.events.submitSuccess || state.events.submitSuccess.length === 0) {
+                    const globals = {
+                        form: self.getModel().getRuleNode(),
+                        event: {
+                            type: action.type,
+                            payload: action.payload,
+                        }
+                    };
+                    FormView.customFunctions.defaultSubmitSuccessHandler(globals);
                 }
             }, "submitSuccess");
             this._model.subscribe((action) => {
                 let state = action.target.getState();
-                let defaultSubmissionError = FormView.LanguageUtils.getTranslatedString(this.getLang(), "InternalFormSubmissionError");
-                window.alert(defaultSubmissionError);
+                // execute the handler only if there are no rules configured on submitError event.
+                if (!state.events.submitError || state.events.submitError.length === 0) {
+                    let defaultSubmissionError = FormView.LanguageUtils.getTranslatedString(this.getLang(), "InternalFormSubmissionError");
```

Comment:
Can you also change this.getLang to self.getLang ? Ideally, we need to use from closure.

Reply:
Done